google/cloud/aiplatform/jobs.py

```diff
@@ -1913,3 +1918,522 @@ def run(
+                Example: "projects/.../locations/.../modelDeploymentMonitoringJobs/456" or
+                "456" when project and location are initialized or passed.
+            project: Optional[str] = None,
+                Optional project to retrieve BatchPredictionJob from. If not set,
+                project set in aiplatform.init will be used.
+            location: Optional[str] = None,
+                Optional location to retrieve BatchPredictionJob from. If not set,
+                location set in aiplatform.init will be used.
+            credentials: Optional[auth_credentials.Credentials] = None,
+                Custom credentials to use. If not set, credentials set in
+                aiplatform.init will be used.
+        """
+        super().__init__(
+            job_name=model_deployment_monitoring_job_name,
+            project=project,
+            location=location,
+            credentials=credentials,
+        )
+        self._gca_resource = self._get_gca_resource(
+            resource_name=model_deployment_monitoring_job_name
+        )
+
+    @classmethod
+    def _parse_configs(
+        cls,
+        objective_configs: Union[
+            model_monitoring.EndpointObjectiveConfig,
+            Dict[str, model_monitoring.EndpointObjectiveConfig],
+        ],
+        endpoint: str,
+        deployed_model_ids: Optional[List[str]] = None,
+    ) -> List[
+        gca_model_deployment_monitoring_job_compat.ModelDeploymentMonitoringObjectiveConfig
+    ]:
+        """Helper function for matching objective configs with their corresponding models
+
+        Args:
+            objective_configs (Union[model_monitoring.objective.EndpointObjectiveConfig,
+                Dict[str, model_monitoring.objective.EndpointObjectiveConfig]):
+                Required. A single config if it applies to all models, or a dictionary of
+                model_id: model_monitoring.objective.EndpointObjectiveConfig if
+                different model IDs have different configs.
+            endpoint (str):
+                Required. A valid endpoint resource name to launch the MDM job on.
+            deployed_model_ids (Optional[List[str]]):
+                Optional. A list of deployed model IDs to apply the objective config to.
+                Note that a model will have a deployed_model_id that is different from the
+                uploaded model ID, and IDs in this list should consist of deployed model IDs
+                on the same endpoint passed in the argument. If `objective_configs` is a dictionary,
+                then this parameter is ignored. If `objective_configs` is an instance of
+                `model_monitoring.EndpointObjectiveConfig` and `deployed_model_ids` is a non-empty
+                list of valid IDs, then the same objective config will apply to all models in this list.
+
+        Returns:
+            An array of ModelDeploymentMonitoringObjectiveConfig objects
```

Comment:
list, arrays are a different type in Python: https://docs.python.org/3/library/array.html